include/zephyr.h

```diff
@@ -17,4 +17,23 @@
 
 #include <kernel.h>
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Optional application main entry point
+ *
+ * A Zephyr application may optionally define a main() function which will
+ * be called by the kernel after it completes the initialization.
+ * Such a function must take no arguments and return no value.
+ *
+ * This function may return but does not have to.
```

Comment:
Do we define what happens If the main function does return, e.g., the system will shutdown/power off?

Reply:
There's no special semantics currently, it will just cause the main thread to exit.

Reply:
@dbkinder Would you like me to either link to that [part of the documentation](https://docs.zephyrproject.org/latest/reference/kernel/threads/system_threads.html) or replicate in this comment part of it? (It's a bit unclear to me how you would like to proceed)

Reply:
@andrewboie @aescolar @nashif  I guess I was looking for a defined behavior of the system if the main function returns and that thread is terminated. Architecturally, what code does the processor execute after that then and what will the system be doing?  If that's not defined by Zephyr, then so be it.

Reply:
another ready thread will then run, which may be the idle thread if no other threads are present or ready to run.

main is just another thread. nothing special happens when it returns.

Reply:
if all threads exit then the system just sits there in the idle thread for each cpu.

Reply:
Ah.  Thanks for clarifying that for me @andrewboie, maybe it's obvious that there's always an idle thread around for a CPU to run.